runtime/lua/vim/treesitter.lua

```diff
@@ -75,6 +75,11 @@ function M.create_parser(bufnr, lang, id)
   if bufnr == 0 then
     bufnr = a.nvim_get_current_buf()
   end
+
+  -- This is required to generate a tree in sync with the buffer.
+  -- We call this anyway because calling this on an already loaded buffer is a noop
```

Comment:
You can just remove this comment. It isn't exactly true anyway, since it checks if there were changes to the file and some other stuff I think. But it doesn't matter. Just load it is good :)

Reply:
doc says
```
 Ensure the buffer {expr} is loaded.  When the buffer name
                refers to an existing file then the file is read.  Otherwise
                the buffer will be empty.  If the buffer was already loaded
                then there is no change.
```
Is that not correct?

Reply:
Looking at the C code, `bufload` actually does the same check as `bufloaded` before actually loading the file.